tests/components/humidifier/test_device_condition.py

```diff
@@ -0,0 +1,314 @@
+
+    hass.states.async_set(
+        "humidifier.entity", STATE_ON, {const.ATTR_MODE: const.MODE_HOME}
+    )
+
+    # Should not fire
+    hass.bus.async_fire("test_event3")
+    await hass.async_block_till_done()
+    assert len(calls) == 3
+
+
+async def test_capabilities(hass):
+    """Test capabilities."""
+    hass.states.async_set(
+        "humidifier.entity",
+        STATE_ON,
+        {
+            const.ATTR_MODE: const.MODE_AWAY,
+            const.ATTR_AVAILABLE_MODES: [const.MODE_HOME, const.MODE_AWAY],
+        },
+    )
+
+    # Test mode
+    capabilities = await device_condition.async_get_condition_capabilities(
+        hass,
+        {
+            "condition": "device",
+            "domain": DOMAIN,
+            "device_id": "",
+            "entity_id": "humidifier.entity",
+            "type": "is_mode",
+        },
+    )
+
+    assert capabilities and "extra_fields" in capabilities
+
+    assert voluptuous_serialize.convert(
+        capabilities["extra_fields"], custom_serializer=cv.custom_serializer
+    ) == [
+        {
+            "name": "available_modes",
+            "options": [("home", "home"), ("away", "away")],
+            "required": True,
+            "type": "select",
+        }
+    ]
+
+
+async def test_capabilities_no_state(hass):
+    """Test capabilities while state not available."""
+    # Test mode
+    capabilities = await device_condition.async_get_condition_capabilities(
+        hass,
+        {
+            "condition": "device",
+            "domain": DOMAIN,
+            "device_id": "",
+            "entity_id": "humidifier.entity",
+            "type": "is_mode",
+        },
+    )
+
+    assert capabilities and "extra_fields" in capabilities
+
+    assert voluptuous_serialize.convert(
+        capabilities["extra_fields"], custom_serializer=cv.custom_serializer
+    ) == [
+        {"name": "available_modes", "options": [], "required": True, "type": "select"}
+    ]
+
+
+async def test_get_condition_capabilities(hass, device_reg, entity_reg):
+    """Test we get the expected toggle capabilities."""
+    config_entry = MockConfigEntry(domain="test", data={})
+    config_entry.add_to_hass(hass)
+    device_entry = device_reg.async_get_or_create(
+        config_entry_id=config_entry.entry_id,
+        connections={(device_registry.CONNECTION_NETWORK_MAC, "12:34:56:AB:CD:EF")},
+    )
+    entity_reg.async_get_or_create(DOMAIN, "test", "5678", device_id=device_entry.id)
+    expected_capabilities = {
+        "extra_fields": [
+            {"name": "for", "optional": True, "type": "positive_time_period_dict"}
+        ]
+    }
+    conditions = await async_get_device_automations(hass, "condition", device_entry.id)
+    for condition in conditions:
+        capabilities = await async_get_device_automation_capabilities(
+            hass, "condition", condition
+        )
+        assert capabilities == expected_capabilities
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)